FAI-362: Score Cards: Integrate with VSCode channel

manstis opened the pull request:
See https://issues.redhat.com/browse/FAI-362

@tiagobento I'm going to need some help here please.

![Screenshot from 2021-02-03 11-04-43](https://user-images.githubusercontent.com/497544/106745536-22438d00-6619-11eb-8cb5-c0a13a54bf79.png)

**css**

The PMML Editor "host" page used for the "showcase" webapp does not have any CSS links/references. PatternFly seems to inject the CSS into the "host" HTML page programmatically on load or something. Come to mention it the PMML editor does not have an `index.html` page itself and only exists as a React component.

![Screenshot from 2021-02-03 12-14-32](https://user-images.githubusercontent.com/497544/106745770-7ea6ac80-6619-11eb-97ae-40f97203005d.png)

It appears as though it is not being injected into the HTML page created in `vscode-extension/KogitoEditor.setupWebviewContent()`. Do you have any ideas how I can solve this? 

**Monaco web-worker**

The web-worker required for Monaco auto-complete does not appear to be loaded either. Possibly due to the same reason as CSS above. DMN uses Monaco too. Is there anything specific that needs to be done for VSCode? Running the PMML Editor showcase from `pmml-editor/dist/index.html` has the webworker running fine (but I had to allow the browser to load resources from _file_ paths).

**Keyboard shortcut bindings**

I was expecting undo/redo from the VSCode toolbar to work "out of the box". The PMML Editor implements `Editor.undo()` and `Editor.redo()`. Am I missing something? The PMML Editor does not have any other keyboard shortcuts (outside of _normal_ browser/HTML bindings, e.g. tab, enter etc).

**window.confirm(..)**

The PMML Editor uses `window.confirm(....)`. This appears not to work inside VSCode, but I suspect this is by (Microsoft's) design and hence we'd need to move to PatternFly modals. Can you please confirm?

**Notification API**

Is this merged yet? How would I bind to it from a React editor?

ederign commented:
Hi @manstis, @tiagobento will reply to the other questions, but meanwhile, a quick note about Alerts API.

The PR is approved and it will be merged soon. We are just waiting for the 0.8.3 release (early next week). As soon as it gets merged, @adrielparedes will drop an email about usage/integration and it will also write a blog post.

manstis commented:
Thanks @ederign. 

@tiagobento has said he'd have a look tomorrow.

manstis commented:
@tiagobento Good news (of sort) for the CSS issue.

It is caused by VSCode adding "dark-theme" to the WebView used to host the editor:

The IFRAME's document's `<body>` element has `class="vscode-dark"`:

![Screenshot from 2021-02-05 12-45-24](https://user-images.githubusercontent.com/497544/107035700-78970400-67b0-11eb-9b38-ccde30896fca.png)

Removing that manually and the editor renders perfectly:-

![Screenshot from 2021-02-05 12-45-32](https://user-images.githubusercontent.com/497544/107035708-7b91f480-67b0-11eb-86a3-d9cbf5736bbd.png)

Is there anything we can do about that!?!

tiagobento commented:
@manstis Hi Michael! First of all thank you for sending this PR! It's great to see a brand-new, React-based Editor come to life and be part of our development VS Code extension. Also sorry for taking so long to review this PR, but I felt like I needed to do it properly :)

The code looks fine and it seems to be that you're already very familiar with the Multiplying Architecture concepts! I was able to build your branch and create a new PMML file directly on VS Code, but the issues you described were present. Here are my replies to your questions, hope it helps!

> css

So PatternFly gets included on the Editors webview because we use PatternFly to display two things that are common for all Editor envelopes - Loading Screen and Keyboard Shortcuts panel. On your `PMMLEditorEnvelopeApp` file you're building the "app" that'll run inside the Webview. Since the PMML Editor is a library, I believe that you should configure it's webpack rules to exclude everything that's coming from node_modules. See `webpack.config.js` on `kie-bc-editors`. This will make the `pmml-editor` be bundled without the libraries. These libraries will then be bundled when `PMMLEditorEnvelopeApp` is built. Also, because of the `dark-theme` and because you're using PF4 already, I believe you should just make the PMML Editor be compatible with PF4's dark theme. This switch is on https://github.com/kiegroup/kogito-tooling/blob/master/packages/editor/src/envelope/styles.scss. You can notice that on the DMN/BPMN/SceSim Editors, the Loading Screen and the Keyboard Shortcuts panel both react to the current theme on VS Code.

> Monaco web-worker

All external assets required by the Editors should be injected dynamically on the page on the EditorFactory. See https://github.com/kiegroup/kogito-tooling/blob/master/packages/kie-bc-editors/src/GwtEditorMapping.ts and https://github.com/kiegroup/kogito-tooling/blob/master/packages/kie-bc-editors/src/GwtEditorWrapperFactory.ts#L111-L113.

If it's possible for you, you can also use our [Standalone Monaco Editor](https://preview.npmjs.com/package/@kiegroup/monaco-editor) from NPM directly. If you wan't to use the default Monaco bundle from Microsoft, just make sure that you include all the static assets on the VS Code bundle. You'll probably need to copy it manually from node_modules using Webpack's Copy Plugin.

> Keyboard shortcut bindings

Looks like changes on the PMML Editor are not being communicated to VS Code, since the file is never changing to `dirty` state. To do that, you need send the `receive_newEdit` notification to the Channel upon each change on the Editor.

> window.confirm(..)

Found this: https://github.com/Microsoft/vscode/issues/67109.

Looks like `window.alert` and `window.confirm` are disabled on WebViews when `enableScripts` is true on the Webview configuration. Which is our case. The VS Code maintainer advised to use VS Code's Quick Input API, but it's too specific, so as I commented on #406, I believe going with your custom-made PF4 Modals is a much better solution. Even more now that we know that `window.alert` and `window.confirm` don't work on VS Code. 

> Notification API

It is merged on `appformer` and `kie-wb-common`, but we have not updated `kogito-tooling` yet, since we're waiting to branch the 0.8.3 release. When it's available to use we'll communicate it broadly. cc @adrielparedes

manstis commented:
Hi @tiagobento 

I've pushed a new commit with some fixes; but some issues still elude me:

1) CSS

I've added your proposed fix; but I still get interference from VSCode's Dark Theme.

2) Monaco web-worker

Fixed.

I use a different Monaco NPM bundle: https://preview.npmjs.com/package/react-monaco-editor

This already bundles everything "internally" with https://github.com/Microsoft/monaco-editor-webpack-plugin

I've added the plugin to `webpack` configuration for PMML in `vscode-extension-pack-kogito-kie-editors` and it's now OK.

3) Keyboard shortcut bindings

Fixed. Thank-you for the pointer.

4) `window.confirm(..)`

Fixed. Simply removed for now until we, the PMML guys, decide what we're doing with Modals.

We use PF Modals for other things at the moment and we want to avoid the use of nested Modals.

manstis commented:
@sjamboro I've added basic Integration Tests to this PR too.

They are based on those already present for the other editors written by @jomarko.

manstis commented:
@tiagobento @ederign 

This PR is ready for review and merge....

However I am conscious you may not want to merge it yet:-
- It **will** add PMML to VSCode
- IDK if you need to make other changes elsewhere to infrastructure/extension stores etc?
- `pmml-editor` is not published to `npm` at the moment (Can it? We have little unit tests at present)
-  The issue relating to PF3/PF4 styling remains. See my email to you both.

I will keep this as "draft" until we are clearer on how to proceed.

tiagobento commented:
@manstis So IIUC there's only one issue remaining here and it is the CSS. As you found out, there's no officially-supported Dark Theme on PF4 yet, and because we're "incorporating" VS Code's Dark Theme, we're facing some issues on our components, that are left in a weird Light-Dark Theme mix.

Since VS Code's Dark Theme is being incorporated at the Editor Envelope level, one alternative is to change the Editor Envelope itself. Another fix can be adding specific CSS to the PMML Editor Envelope so that we fix the Dark Theme-related issues one by one. IDK what is the preferred way for you guys, though. Having the PMML Editor start with a Dark Theme ready out-of-the-box might be nice, but maybe it'll require more work too. Please tell me how do you want to proceed so that we make a plan to potentially remove the Dark Theme on the Editor Envelope.

tiagobento commented:
Btw, really nice finding about the Monaco Editor. This is probably going to be very useful for the DMN and BPMN Editors as well!

manstis commented:
@tiagobento Before this is merged I want to get all the other PRs for PMML merged first and probably complete the validation refactoring I've mentioned before. It's pointless publishing this to Marketplace knowing there is so much in the pipeline.

manstis commented:
@tiagobento re: the CSS problem.

> So IIUC there's only one issue remaining here and it is the CSS. As you found out, there's no officially-supported Dark Theme on PF4 yet, and because we're "incorporating" VS Code's Dark Theme, we're facing some issues on our components, that are left in a weird Light-Dark Theme mix.

Not quite.

The envelope's "Keyboard shortcut bar, icon and modal" although written as part of `kogito-tooling` referencing PF4 is actually using the GWT Editor's PF3 CSS. If you run the PMML editor these components are rendered with _light_ styling (even though VSCode may be running with a dark theme enabled). If you run the DMN editor these components are rendered with a _dark_ theme HOWEVER if you remove the `link` in the `<head>` of the DMN editor for the PF3 CSS then the "Keyboard shortcut etc" components are rendered as they are for PMML i.e. with PF4 CSS and light styling.

So.. the problem is the "Keyboard shortcut" stuff although looking like it uses PF4 CSS is actually using PF3 CSS.

You might want to remove the "Keyboard shortcut" components styling dependency on PF all together?

manstis commented:
Oh, I also want to integrate the Notifications API with this PR.....

tiagobento commented:
@manstis Hmm.. interesting. The Keyboard Shortcuts panel and the Loading screen both use PF4 React Components. That's why we depend on PF4 and that's why PF4 is included on the Editor Envelopes by default. From your investigation I understand that PF3 and PF4 are conflicting there in some way, leaving things "working by chance", which is not great. We'll analyse all that and try to figure out a good solution. In the meantime, we can talk about the Notifications API. cc @adrielparedes 

Basically, there are new methods available on the `KogitoEnvelopeChannelApi` that you can call to display notifications to users (both as a popup at the bottom-right and in the Problems tab in VS Code). The API is pretty simple so I'll let you explore and reach out if you have questions! Also, there's a new method called `validate` on the Editor interface. This method is called 
every time before saving, so you can effectively block the Save operation by returning a non-empty list of Notifications on the `validate` method. Note that the Notifications API is still no-op for channel other than VS Code.

tiagobento commented:
@manstis No problem about waiting for other PRs. We still have issues to fix on this one, so all good.

manstis commented:
@tiagobento 

> @manstis Hmm.. interesting. The Keyboard Shortcuts panel and the Loading screen both use PF4 React Components. That's why we depend on PF4 and that's why PF4 is included on the Editor Envelopes by default. From your investigation I understand that PF3 and PF4 are conflicting there in some way, leaving things "working by chance", which is not great. We'll analyse all that and try to figure out a good solution. 

Yes, IMO it is "working by chance". The envelope/Keyboard Shortcuts are using PF4 components but are picking up PF3 CSS from the CSS added to the HEAD by the GWT editor configuration... if you remove the PF3 CSS link from the editors' `WebView` the keyboard shortcuts for DMN/BPMN will look exactly like they do for PMML (i.e. PF4 styling.. light only theme).

> In the meantime, we can talk about the Notifications API. cc @adrielparedes 
> Basically, there are new methods available on the `KogitoEnvelopeChannelApi` that you can call to display notifications to users (both as a popup at the bottom-right and in the Problems tab in VS Code). The API is pretty simple so I'll let you explore and reach out if you have questions! Also, there's a new method called `validate` on the Editor interface. This method is called
> every time before saving, so you can effectively block the Save operation by returning a non-empty list of Notifications on the `validate` method. Note that the Notifications API is still no-op for channel other than VS Code.

This PR is now fully updated to integrate with `StateControl` and `Notifications` :-)

adrielparedes commented:
great!! love to see that working :)

caponetto commented:
FYI: I'm investigating the CSS issue :)

manstis commented:
@caponetto Thank-you. 

My investigations revealed that the GWT editors inject PF3 CSS into the host/envelope page as part of their defined resources. The keyboard shortcut components use class names _thinking_ they're PF4 as that's their dependency however the GWT editors injection of PF3 CSS is taking precedence as it's the last to define the class definitions.

manstis commented:
@caponetto I am happy to demonstrate my investigation if it helps give you a head start.

jomarko commented:
@manstis hello, we met after some time on PR, yay.

As you know, we consume PMML files also in DMN, so I would like to ask, what is the purpose of this PR.

Will be users able to:
- open any PMML file using VSCode?
- open just some specific PMML files using VSCode?
- or it is not read only, is user also able to create and save some PMML files?

manstis commented:
Hi @jomarko we do indeed (virtually) meet :-)

This PR adds limited PMML support to VSCode.
- The User will be able to open ANY PMML 4.4 file
- The User will only be able to create or edit Scorecards
- The User will be able to view Linear Regression Models
- All other model types are simply dumped as formatted JSON in the editor for viewing.

caponetto commented:
Hi @manstis, I'm still investigating the fundamental issue but I guess at least I managed to converge the css.

![Peek 2021-03-05 16-50](https://user-images.githubusercontent.com/638737/110166505-3bd02400-7dd3-11eb-87f3-7a46300edacd.gif)

manstis commented:
@tiagobento @ederign @jomarko @kelvah @tomasdavidorg This PR is ready for your reviews.

@jomarko You became "added" IDK if you want/need to review in addition to @tomasdavidorg. The choice is yours.

tomasdavidorg commented:
@manstis 
@jomarko will take a look on this 🙂

manstis commented:
@tiagobento All checks are green.. don't anybody dare ask to a change ;-)

jomarko commented on packages/vscode-extension-pack-kogito-kie-editors/src/ui-test/helpers/PmmlEditorTestHelper.ts:
```diff
@@ -0,0 +1,128 @@
+   * @returns a promise resolving to WebElement of opened modal.
+   */
+  public openDataDictionary = async (): Promise<Modal> => {
+    const button = await this.getDataDictionaryButton();
```
Could we please make all `get*Button()` methods private?

manstis replied:
Done.

jomarko commented on packages/vscode-extension-pack-kogito-kie-editors/src/ui-test/helpers/PmmlEditorTestHelper.ts:
```diff
@@ -0,0 +1,128 @@
+    await assertWebElementIsDisplayedEnabled(button);
+    await button.click();
+
+    const body = await this.webview.findWebElement(By.xpath("//div[@data-title='DataDictionaryModal']"));
+    const close = await this.webview.findWebElement(By.xpath("//button[@data-title='DataDictionaryModalClose']"));
+    return Promise.resolve(new Modal(body, close));
```
Could we please introduce private method?
```ts
  private resolveModal = async (dataTitle: string): Promise<Modal> => {
    const body = await this.webview.findWebElement(By.xpath(`//div[@data-title='${dataTitle}']`));
    const close = await this.webview.findWebElement(By.xpath(`//button[@data-title='${dataTitle}Close']`));
    return Promise.resolve(new Modal(body, close));
  };
```

and apply it three times?
- `return this.resolveModal("DataDictionaryModal");`
- `return this.resolveModal("MiningSchemaModal");`
- `return this.resolveModal("OutputsModal");`

Feel free to rename `resolveModal` to anything more meaningful.

tiagobento replied:
Hi @jomarko! Although I understand where this request came from, I'd like to add that I disagree that extracting a method is the best thing to do in this case. Tests are supposed to be self-contained, self-explanatory descriptions of the use-case being tested, and adding indirections makes it harder to follow the use-case, IMHO.

More than that, when we start using too many "utility" methods on simple tests, we start creating indirections that can potentially evolve to complex execution trees with a bunch of methods reused.

For test cases, specifically, I'm much more inclined to keep the "duplication" (which is not even that bad here), instead of creating an abstraction/indirection.

jomarko replied:
@tiagobento hello, I do not agree with these points:
- `be self-contained` I am not asking to introduce a public method in a framework, but a private method in the same file
- `self-explanatory descriptions` from my point of view proper method name (what I asked Michael to choose the best to fits his needs) and the parameter , in this case class name, what Michael designed very well, are on the same level of self-explain-ability as the current solution

I am ok with your proposal, if Michael confirms, he implemented three modals by three different classes in typescript/react. If there is chance, xpaths we are using will need to be updated just for one of three cases. Saying differently, if there is a chance `div[@data-title='DataDictionaryModal']` will change to `canvas[@data-title='DataDictionaryModal']` but we keep `div[@data-title='MiningSchemaModal']` and `div[@data-title='OutputsModal']`, then I agree with you. duplication is fine, otherwise I require suggested changes.

manstis replied:
Done as requested @jomarko.

jomarko reviewed (commented):
@manstis thank you for PR. Some comments/questions to code inline.

And here are my questions coming from manual review.
- Do we have some ticket to upgrade extension description shown when installing to vs code? Currently there is no note about PMML
- I wasn't able to save 'Is Scorable' and 'Use Reason Code' from model setup, after reopening file always set to true, is it expected?
- When opening an invalid pmml file, never ending loading screen is shown, is that tracked by some ticket please?

jomarko commented:
@manstis https://github.com/kiegroup/kogito-tooling/pull/409#issuecomment-799286837 - sorry, it doesn't work like this.

kelvah commented on packages/pmml-editor/src/editor/history/HistoryProvider.ts:
```diff
@@ -79,15 +83,17 @@ export class HistoryService {
     const newState: M = produce(state, recipe, (patches, inversePatches) => {
       this.history.changes.push({ path: path, change: patches, reverse: inversePatches });
+      this.history.index = this.history.changes.length;
+      this.listeners.forEach(listener => listener(`Command${this.history.index}`));
```
I see you are calling the listeners inside `mutate`, I am wondering if this should be done inside `commit` instead, after the changes have been validated.

kelvah commented on packages/pmml-editor/src/editor/styles.scss:
```diff
@@ -0,0 +1,28 @@
+ * limitations under the License.
+ */
+
+@import "../../../../node_modules/@patternfly/patternfly/sass-utilities/all";
+@import "../../../../node_modules/@patternfly/patternfly/base/variables";
+@import "../../../../node_modules/@patternfly/patternfly/base/themes";
+
+.pf-c-button.pf-m-secondary {
+  color: var(--pf-global--primary-color--100);
+}
+
+.pf-c-button.pf-m-primary {
+  color: var(--pf-global--Color--light-100);
+  background-color: var(--pf-global--primary-color--100);
```
I think the overrides here aren't necessary anymore. Incidentally they are causing issues with buttons in disabled state.

manstis replied:
Removed.

tiagobento commented:
@manstis As we spoke today, the 0.8.4 release won't include this PR. For this reason, can you please rebase it with `master` and change `.github/workflows/publish_artifacts.yml` to not exclude the `@kogito-tooling/pmml-editor`? This way, when this PR is in, we will be certainly publishing the PMML Editor package to NPM on the next release.

manstis commented:
@jomarko Thank-you for your review:

> Do we have some ticket to upgrade extension description shown when installing to vs code? Currently there is no note about PMML

Fixed.

> I wasn't able to save 'Is Scorable' and 'Use Reason Code' from model setup, after reopening file always set to true, is it expected?

Fixed.

> When opening an invalid pmml file, never ending loading screen is shown, is that tracked by some ticket please?

Thank-you for reporting. See https://issues.redhat.com/browse/FAI-442

manstis commented on packages/pmml-editor/src/__tests__/editor/history/HistoryProvider.test.ts:
```diff
@@ -16,7 +16,10 @@
+const listener1 = jest.fn();
+const listener2 = jest.fn();
+
+const service: HistoryService = new HistoryService([listener1, listener2]);
```
`listeners` are not needed for those tests. Thank-you for highlighting. 

I now pass an empty array as you proposed.

manstis commented on packages/pmml-editor/src/editor/styles.scss:
```diff
@@ -0,0 +1,28 @@
+}
+
+.pf-c-button.pf-m-primary {
+  color: var(--pf-global--Color--light-100);
+  background-color: var(--pf-global--primary-color--100);
```
File no longer required. Removed.

manstis commented on packages/pmml-editor/src/editor/validation/NotificationsFactory.ts:
```diff
@@ -0,0 +1,30 @@
+    return {
+      path: path,
+      message: validationEntry.message ?? "",
+      type: "PROBLEM",
+      severity: validationEntry.level === ValidationLevel.ERROR ? "ERROR" : "WARNING"
```
Fixed as proposed.

manstis commented on packages/pmml-editor/src/editor/components/DataDictionary/DataDictionaryHandler/DataDictionaryHandler.tsx:
```diff
@@ -57,14 +57,15 @@ const DataDictionaryHandler = () => {
-        }
-      });
-    }
+    //TODO {manstis} How are we to handle channel-agnostic confirmations?
+    //if (window.confirm(`Delete Output "${dictionary[index].name}"?`)) {
```
Done. I changed the comments to reference https://issues.redhat.com/browse/FAI-443

manstis commented on packages/pmml-editor/src/editor/components/EditorScorecard/molecules/CharacteristicsTableEditRow.tsx:
```diff
@@ -152,16 +152,17 @@ export const CharacteristicsTableEditRow = (props: CharacteristicsTableEditRowPr
-          }
-        });
-      }
+      //TODO {manstis} How are we to handle channel-agnostic confirmations?
+      //if (window.confirm(`Delete Attribute?`)) {
```
Same answer (and there were a few other places too).

manstis commented on packages/pmml-editor/src/editor/PMMLEditor.tsx:
```diff
@@ -40,13 +38,18 @@ import { EmptyStateNoContent } from "./components/LandingPage/organisms";
 interface Props {
   exposing: (s: PMMLEditor) => void;
-  channelApi: MessageBusClientApi<KogitoEditorChannelApi>;
+  ready: () => void;
+  newEdit: (edit: KogitoEdit) => void;
+  setNotifications: (path: string, notifications: Notification[]) => void;
```
Done.

manstis commented:
@tiagobento `pmml-editor` is now not excluded from `npm` publication...

This PR is ready for your review (Jozef and Roberto have had their first pass.. I am hoping there is little else to correct).

jomarko reviewed (approved):
thank you @manstis

kelvah reviewed (approved):
:ok_hand:

manstis commented:
@tiagobento Only your review remains.. any idea why the 3 VSCode integration tests keep failing?

The logs only suggest `warning vscode-extension-pack-kogito-kie-editors@0.8.4: The engine "vscode" appears to be invalid.`

caponetto commented:
@manstis This might help (log artifact uploaded by the build):
```xml
<testcase classname="Editors are loading properly" name=""after each" hook for "Opens demo.pmml file in PMML Editor"" time="0.022">
<failure>no such element: Unable to locate element: {"method":"css selector","selector":".monaco-workbench"} (Session info: chrome=87.0.4280.141) (Driver info: chromedriver=87.0.4280.88 (89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761}),platform=Linux 5.4.0-1040-azure x86_64) NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":".monaco-workbench"} (Session info: chrome=87.0.4280.141) (Driver info: chromedriver=87.0.4280.88 (89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761}),platform=Linux 5.4.0-1040-azure x86_64) at Object.checkLegacyResponse (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/error.js:546:15) at parseHttpResponse (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/http.js:509:13) at /home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/http.js:441:30 at runMicrotasks (<anonymous>) at processTicksAndRejections (internal/process/task_queues.js:97:5) From: Task: WebDriver.findElement(By(css selector, .monaco-workbench)) at Driver.schedule (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:807:17) at Driver.findElement (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:1014:17) at new AbstractElement (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/AbstractElement.js:33:43) at new EditorView (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:25:9) at VSCodeTestHelper.<anonymous> (out/ui-test/helpers/VSCodeTestHelper.js:137:50) at step (out/ui-test/helpers/VSCodeTestHelper.js:48:23) at Object.next (out/ui-test/helpers/VSCodeTestHelper.js:29:53) at /home/runner/work/kogito-tooling/kogito-tooling/packages/vscode-extension-pack-kogito-kie-editors/out/ui-test/helpers/VSCodeTestHelper.js:23:71 at new Promise (<anonymous>) at __awaiter (out/ui-test/helpers/VSCodeTestHelper.js:19:12) From: Task: WebElement.findElement(By(css selector, *[id="workbench\.parts\.editor"])) at Driver.schedule (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:807:17) at WebElementPromise.schedule_ (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:2010:25) at WebElementPromise.findElement (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:2057:17) at new AbstractElement (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/AbstractElement.js:39:48) at new EditorView (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:25:9) at VSCodeTestHelper.<anonymous> (out/ui-test/helpers/VSCodeTestHelper.js:137:50) at step (out/ui-test/helpers/VSCodeTestHelper.js:48:23) at Object.next (out/ui-test/helpers/VSCodeTestHelper.js:29:53) at /home/runner/work/kogito-tooling/kogito-tooling/packages/vscode-extension-pack-kogito-kie-editors/out/ui-test/helpers/VSCodeTestHelper.js:23:71 at new Promise (<anonymous>) From: Task: WebElement.findElements(By(css selector, .editor-group-container)) at Driver.schedule (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:807:17) at EditorView.schedule_ (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:2010:25) at EditorView.findElements (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/selenium-webdriver/lib/webdriver.js:2081:19) at EditorView.<anonymous> (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:137:41) at Generator.next (<anonymous>) at /home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:8:71 at new Promise (<anonymous>) at __awaiter (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:4:12) at EditorView.getEditorGroups (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:136:16) at EditorView.<anonymous> (/home/runner/work/kogito-tooling/kogito-tooling/node_modules/vscode-extension-tester/node_modules/monaco-page-objects/out/components/editor/EditorView.js:58:37)</failure>
</testcase>
```

tiagobento commented:
@manstis Thanks for fixing the tests! Waiting for a green build then this PR can be merged.

tiagobento commented:
All green! Thank you for everyone involved on this PR. Great work!
